scripts/validate-build.js

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env node
+// eslint-disable-next-line strict
+'use strict';
+
+const fs = require('fs');
+const esprima = require('esprima');
+const walk = require('esprima-walk');
+const colors = require('colors');
+
+/**
+ * Load the Abstract Syntax Tree from the javascript file given
+ *
+ * @param {String} file - The path to the javascript filename
+ * @returns {ASTTree} Returns the AST tree
+ */
+function loadAST(file) {
+  try {
+    return esprima.parse(
+      String(fs.readFileSync(file)),
+      { comment: true }
+    );
+
+  } catch (e) {
+    console.error('Unable to parse source:', e);
+
+    return false;
+  }
+}
+
+/**
+ * Collect all comments from the AST
+ *
+ * @param {ASTTree} ast - The AST tree
+ * @returns {Array<ASTNode>} Returns the comment AST nodes
+ */
+function loadComments(ast) {
+  let comments = [];
+
+  walk(ast, function (node) {
```

Comment:
@wavesoft The behavior has been like this across all versions of Esprima.

To illustrate _comment collection_, run this with Node.js REPL (assuming `esprima` module is installed):

```js
> var esprima = require('esprima')
> esprima.parse('x + y // hello', { comment: true })
Program {
  type: 'Program',
  body: [ ExpressionStatement { type: 'ExpressionStatement', expression: [Object] } ],
  sourceType: 'script',
  comments: [ { type: 'Line', value: ' hello' } ] }
```
Only the top-level node _Program_ will ever have the `comments` array. You can prove this by traversing the node top-to-bottom, no other node will have such an array.

This is not to be confused with _comment attachment_, as demonstrated below:

```js
> var esprima = require('esprima')
> esprima.parse('x + y // hello', { attachComment: true })
Program {
  type: 'Program',
  body: [ ExpressionStatement { type: 'ExpressionStatement', expression: [Object] } ],
  sourceType: 'script' }
> esprima.parse('x + y // hello', { attachComment: true }).body[0].expression
BinaryExpression {
  type: 'BinaryExpression',
  operator: '+',
  left: Identifier { type: 'Identifier', name: 'x' },
  right: Identifier { type: 'Identifier', name: 'y' },
  trailingComments: [ { type: 'Line', value: ' hello', range: [Object] } ] }
```
Look at how `trailingComments` array exists only for the _BinaryExpression_ node (because that particular comment is attached to the node).

I believe that in your particular use case, what you want is to _collect_ every single comment, not to have it attached to various node. Therefore, it is sufficient to inspect the top-level Program node and grab the `comments` array and look at its element, **without** the need to traverse the whole AST.

Does this make sense?